third-culture-software/bhima
# Implement aggregate stock consumption per patient report

This report shows the aggregate stock consumption to patients and services.

Note:  Columns 3-7 are combined data for distributions to patients or services.   The last 4 columns are for distributions for patients.

Closes https://github.com/IMA-WorldHealth/bhima/issues/5448

It is in the Stock > Reports menu

**TESTING**
- Try it out a date range (without inventory group)
   - Notice the the right 4 columns are blank for "To Service" items.
- Try it out with an inventory group (such as 'Medicaments en comprimes" in IMCK db)
   - Notice new subtitle for this inventory group
   - Notice the change in numbers of results

To verify the tabulated results, pick an
 inventory code from the results above, then
- Run a stock sheet for that inventory item.  You may want to limit the time range to get a smaller set of data to compare.
- Run a Aggregated Stock Report for the same period
- You should be able to correlate the "To Service" and "To Patient" items with the data from the aggregated stock report to validate the new report

This SQL query might be useful:

[check_agg.sql.txt](https://github.com/IMA-WorldHealth/bhima/files/6598052/check_agg.sql.txt)
